### ISSUE TITLE
EPIC: zwei Karten über NVLink nutzen — parallele Läufe und grössere Modelle

### ISSUE THREAD

**Issue by thodel** [thodel/training-atr-models#12]
## Wofür der Umzug gemacht wird

asteraix hat **zwei A40 à 46 GB, beide leer**, über **NVLink** verbunden — vier Links à 14,06 GB/s, P2P aktiv (gemessen 16.09.2026):

```
        GPU0   GPU1
GPU0     X     NV4
GPU1    NV4     X
P2P:    OK
```

Auf idhefix teilt sich ein Lauf **31,6 GB** mit den Serving-Diensten. Hier stehen **92 GB** zur Verfügung. Das ist der Grund für die Trennung, und es ist heute aus drei Gründen nicht nutzbar:

1. **`max_concurrent: 1`** in `TrainerSettings` — der Scheduler startet grundsätzlich einen Job.
2. **Alle fünf Units pinnen `CUDA_VISIBLE_DEVICES=1`**, und `params.device` ist `cuda:0`. Der Scheduler *setzt eine Karte voraus*, statt eine zuzuteilen.
3. **Die VRAM-Vorprüfung ist global**, nicht je Karte (`preflight.py`).

## Ausdrücklich nach dem Cutover

Dieser Epic ist **kein Migrationsschritt**. Wer ihn mit #1 vermischt, kann hinterher nicht sagen, ob ein anderes Ergebnis vom Umzug kommt oder vom Umbau. Erst zieht v5 um und liefert eine vergleichbare Zahl (#10), dann wird ausgebaut.

## Shape

- **Kartenzuteilung**: der Scheduler vergibt eine Karte, statt `CUDA_VISIBLE_DEVICES` vorauszusetzen. `max_concurrent` wird eine Funktion der freien Karten
- **VRAM-Vorprüfung je Karte** statt global
- **Ein Modell über beide Karten**: NVLink macht 92 GB nutzbar. Qwen3-VL-8B wäre damit **ohne 4-bit** trainierbar. Das ist eine **Messung**, keine Annahme — und die erste, die sich lohnt
- **Hyperparameter neu bestimmen.** `batch_size: 1`, `accumulate_grad_batches: 16`, `--load-in-4bit`, `max_seq_len: 4096` sind auf eine geteilte Karte abgestimmt. Auf 46 oder 92 GB sind sie vermutlich zu vorsichtig — und `max_seq_len: 4096` schneidet heute Proben ab (v5 meldete „a sample tokenized to 4515 tokens")
- **Sechs Kommentare, die dann falsch sind.** `preflight.py:4-9`, `settings.py:119`, `hf_source.py:8`, `contracts.py:305`, `runner_base.py:1113` begründen Verhalten mit **idhefix'** Hardware. Der prominenteste: `preflight.py`s Begründung „GPU 1 is shared with the serving engines" — genau das hebt der Split auf

## Tests

- `test_two_jobs_run_on_two_cards`
- `test_the_preflight_checks_the_card_it_will_use`
- `test_a_job_never_lands_on_a_card_another_job_holds`
